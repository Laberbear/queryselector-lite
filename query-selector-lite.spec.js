const fs = require('fs').promises;
const Document = require('./query-selector-lite');

describe('queryselectorlite', () => {
  const testCases = [
    {
      file: 'test.html',
      test: [
        'h3',
        '.abcdef',
        '#testid',
      ],
    },
  ];
  for (const testCase of testCases) {
    for (const test of testCase.test) {
      it(`should test querySelector ${test} in file ${testCase.file}`, async () => {
        const doc = new Document();
        await doc.loadFile(testCase.file);
        expect(doc.querySelector(test)).toMatchSnapshot();
      });
      it(`should test querySelectorAll ${test} in file ${testCase.file}`, async () => {
        const doc = new Document();
        await doc.loadFile(testCase.file);
        expect(doc.querySelectorAll(test)).toMatchSnapshot();
      });
      it(`should test querySelectorAll ${test} in file ${testCase.file} pre loaded`, async () => {
        const html = await (await fs.readFile(testCase.file)).toString();
        const doc = new Document(html);
        expect(doc.querySelectorAll(test)).toMatchSnapshot();
      });
    }
  }
  it('should properly use grouping selectrs', async () => {
    const doc = new Document();
    await doc.loadFile('test.html');
    const result = doc.querySelectorAll('div,h3');
    expect(result.length).toBe(5);
    expect(result.length).toMatchSnapshot();
  });
});
