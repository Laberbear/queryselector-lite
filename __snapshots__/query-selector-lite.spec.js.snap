// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryselectorlite should test querySelector #testid in file test.html 1`] = `
Object {
  "attributes": Object {
    "id": "testid",
  },
  "children": Array [],
  "contentEnd": 149,
  "contentStart": 143,
  "contents": "adsdasd",
  "name": "h2",
  "nameDone": true,
  "tag0End": 142,
  "tag0Start": 127,
  "tag1End": 154,
  "tag1Start": 154,
}
`;

exports[`queryselectorlite should test querySelector .abcdef in file test.html 1`] = `
Object {
  "attributes": Object {
    "class": "abcdef",
  },
  "children": Array [],
  "contentEnd": 111,
  "contentStart": 105,
  "contents": "adsdasd",
  "name": "h1",
  "nameDone": true,
  "tag0End": 104,
  "tag0Start": 86,
  "tag1End": 116,
  "tag1Start": 116,
}
`;

exports[`queryselectorlite should test querySelector h3 in file test.html 1`] = `
Object {
  "attributes": Object {},
  "children": Array [],
  "contentEnd": 175,
  "contentStart": 169,
  "contents": "adsdasd",
  "name": "h3",
  "nameDone": true,
  "tag0End": 168,
  "tag0Start": 165,
  "tag1End": 180,
  "tag1Start": 180,
}
`;

exports[`queryselectorlite should test querySelectorAll #testid in file test.html 1`] = `
Array [
  Object {
    "attributes": Object {
      "id": "testid",
    },
    "children": Array [],
    "contentEnd": 149,
    "contentStart": 143,
    "contents": "adsdasd",
    "name": "h2",
    "nameDone": true,
    "tag0End": 142,
    "tag0Start": 127,
    "tag1End": 154,
    "tag1Start": 154,
  },
]
`;

exports[`queryselectorlite should test querySelectorAll .abcdef in file test.html 1`] = `
Array [
  Object {
    "attributes": Object {
      "class": "abcdef",
    },
    "children": Array [],
    "contentEnd": 111,
    "contentStart": 105,
    "contents": "adsdasd",
    "name": "h1",
    "nameDone": true,
    "tag0End": 104,
    "tag0Start": 86,
    "tag1End": 116,
    "tag1Start": 116,
  },
]
`;

exports[`queryselectorlite should test querySelectorAll h3 in file test.html 1`] = `
Array [
  Object {
    "attributes": Object {},
    "children": Array [],
    "contentEnd": 175,
    "contentStart": 169,
    "contents": "adsdasd",
    "name": "h3",
    "nameDone": true,
    "tag0End": 168,
    "tag0Start": 165,
    "tag1End": 180,
    "tag1Start": 180,
  },
]
`;
